### PR TITLE
Addin IFEval + refactoring for reporting flexible metrics

### PR DESCRIPTION
**Pull request by Kipok**
Will probably do multiple rounds of refactoring here, but this should be ok to unblock us adding new benchmarks easily

### CODE REVIEW

File: pipeline/compute_metrics.py [hunk 1]
```diff
@@ -28,94 +28,221 @@
...
-        correct_answer = total_correct / total * 100.0
-        no_answer = 0
-        wrong_answer = (total - total_correct) / total
-    else:
-        raise ValueError(f"Unsupported eval_type {eval_type}")
-    return correct_answer, wrong_answer, no_answer, total
+    return evaluator.get_metrics()
+
+
+math_benchmarks = [
```
Comment (shtoshni) [inline pipeline/compute_metrics.py:220]: Would be good to move these benchmark-related constants to a separate file

File: nemo_skills/evaluation/evaluate_results.py [hunk 2]
```diff
@@ -54,56 +56,104 @@ def __post_init__(self):
...
+            samples = [preprocess_code(json.loads(line)) for line in f]
+        # all changes will be done with a new key "completion", so it's ok to write to the same file
+        with open(jsonl_file, "wt", encoding="utf-8") as f:
+            for sample in samples:
+                f.write(json.dumps(sample) + "\n")
+        eval_config = {
+            "samples": jsonl_file,
+            "base_only": False,
+            "parallel": None,
+            "i_just_wanna_run": False,
```
Comment (shtoshni) [inline nemo_skills/evaluation/evaluate_results.py:83]: "i_just_wanna_run"??

Comment (Kipok) [reply nemo_skills/evaluation/evaluate_results.py:83]: That's evalplus parameter :)

File: pipeline/compute_metrics.py (hunk 1 above)
Comment (Kipok) [reply pipeline/compute_metrics.py:220]: Yeah, eventually, but maybe not in this PR?

File: nemo_skills/evaluation/evaluate_results.py (hunk 2 above)
Comment (shtoshni) [reply nemo_skills/evaluation/evaluate_results.py:83]: Hahaha

File: pipeline/compute_metrics.py (hunk 1 above)
Comment (shtoshni) [reply pipeline/compute_metrics.py:220]: Sure, thing. May be we can add a ToDo

**Comment by shtoshni** [comment]
The PR looks good to me. Thanks for refactoring the eval and metrics part. I have not tested it, but we can merge if it works with your setup.

**Review by shtoshni — APPROVED** [review]
Thanks for refactoring the eval part!

**Pull request merged**